addons/l10n_ar/models/account_move.py

```diff
@@ -293,3 +293,73 @@ def _get_name_invoice_report(self):
         if self.l10n_latam_use_documents and self.company_id.account_fiscal_country_id.code == 'AR':
             return 'l10n_ar.report_invoice_document'
         return super()._get_name_invoice_report()
+
+    def _l10n_ar_amount_and_taxes(self):
+        self.ensure_one()
+        res = {}
+        if self.is_invoice():
+            tax_lines = self.line_ids.filtered('tax_line_id')
+            included_taxes = self.l10n_latam_document_type_id and \
+                self.l10n_latam_document_type_id._filter_taxes_included(tax_lines.mapped('tax_line_id'))
+            if not included_taxes:
+                l10n_latam_amount_untaxed = self.amount_untaxed
+                not_included_invoice_taxes = tax_lines
+            else:
+                included_invoice_taxes = tax_lines.filtered(lambda x: x.tax_line_id in included_taxes)
+                not_included_invoice_taxes = tax_lines - included_invoice_taxes
+                sign = -1 if self.is_inbound() else 1
+                l10n_latam_amount_untaxed = self.amount_untaxed + sign * sum(included_invoice_taxes.mapped('balance'))
+            res['l10n_latam_amount_untaxed'] = l10n_latam_amount_untaxed
+            res['l10n_latam_tax_ids'] = not_included_invoice_taxes
+        else:
+            res['l10n_latam_amount_untaxed'] = False
+            res['l10n_latam_tax_ids'] = [(5, 0)]
```

Comment:
This is the worst name chosen: l10n_latam_tax_ids while these are account.move.lines

Reply:
Changed